api/insight/apiroutes.go

```diff
@@ -179,28 +210,120 @@ func (c *insightApiContext) broadcastTransactionRaw(w http.ResponseWriter, r *ht
 	writeJSON(w, txid, c.getIndentQuery(r))
 }
 
-func (c *insightApiContext) getAddressTxnOutput(w http.ResponseWriter, r *http.Request) {
-	address := m.GetAddressCtx(r)
-	if address == "" {
-		http.Error(w, http.StatusText(422), 422)
-		return
-	}
-	txnOutputs := c.BlockData.ChainDB.GetAddressUTXO(address)
-	writeJSON(w, txnOutputs, c.getIndentQuery(r))
-}
-
 func (c *insightApiContext) getAddressesTxnOutput(w http.ResponseWriter, r *http.Request) {
-	addresses := strings.Split(m.GetAddressCtx(r), ",")
+	addresses := []string{}
+	addrs := m.GetAddressCtx(r)
+	if addrs != "" {
```

Comment:
Not a big deal, but you could early return instead to make the flow clearer (to me at least).

e.g.:

```go
if addrs == "" {
    writeJSON(w, []apitypes.AddressTxnOutput{}, c.getIndentQuery(r))
    return
}
```